lib/mpool/src/mblock.c

```diff
@@ -157,3 +157,20 @@ mpool_mblock_read(struct mpool *mp, uint64_t mbid, const struct iovec *iov, int
 
     return mblock_fset_read(mclass_fset(mc), mbid, iov, iovc, off);
 }
+
+merr_t
+mpool_mblock_clone(struct mpool *mp, uint64_t mbid, off_t off, size_t len, uint64_t *mbid_out)
+{
+    struct media_class *mc;
+    enum hse_mclass mclass;
+
+    if (!mp || !mbid_out)
+        return merr(EINVAL);
+
+    mclass = mcid_to_mclass(mclassid(mbid));
+    mc = mpool_mclass_handle(mp, mclass);
```

Comment:
delete local var `mclass`  and write `mc = mpool_class_handle(mp, mcid_to_mclass(mbid));` ?